src/mame/mr/nl_dribling.cpp

```diff
@@ -0,0 +1,524 @@
+// license:CC0-1.0
+// copyright-holders: Enzo Lombardi
+
+//
+// Netlist for Model Racing Dribbling: derived from the schematics in the Dribbling manual, located here:
+// https://archive.org/details/ArcadeGameManualDribbling
+// LM339 model adapted from here: https://github.com/evenator/LTSpice-Libraries/blob/master/sub/LM339.sub
+#include "netlist/devices/net_lib.h"
+
+// Known issues:
+// In the PARATA netlist the JFET emulation based on MOSFET doesn't work well. The whole netlist has been
+// replaced with a quite similar one, at least to my ears.
+// The output goes through a TDA2003 amplifier which seems to be a band-pass filter as well cutting the frequency
+// generated by the TOS when it's idle. By defining FILTER_TOS the $FF gets translated into no sound since emulated
+// logic chips are free.
+#define FILTER_TOS
+
+#define TTL_74LS86_DIP TTL_7486_DIP
+#define TTL_74LS107_DIP TTL_74107_DIP
+#define TTL_74LS161_DIP TTL_74161_DIP
+#define TTL_74LS164_DIP TTL_74164_DIP
+#define ATTENUATE_FOLLA 200
+#define REPLACE_LM339
+
+#ifdef REPLACE_LM339
+static NETLIST_START(LM339) 
+{
+	// * CONNECTIONS:   NON-INVERTING INPUT
+	// *                | INVERTING INPUT
+	// *                | | POSITIVE POWER SUPPLY
+	// *                | | | NEGATIVE POWER SUPPLY
+	// *                | | | | OPEN COLLECTOR OUTPUT
+	// *                | | | | |
+	// *                1 2 3 4 5
+	LM324_DIP(IC)
+	ALIAS(1, IC.3)
+	ALIAS(2, IC.2)
+	ALIAS(3, IC.4)
+	ALIAS(4, IC.11)
+	ALIAS(5, IC.1)
+	ALIAS(GND, IC.11)
+	NET_C(GND, IC.5, IC.6, IC.10, IC.9, IC.12, IC.13)
+}
+#else
```

Comment:
What’s the intention of replacing the LM339 with an LM324?

The LM339 is a quad comparator – it’s designed to implement simple voltage comparison, and has single-ended open collector output (i.e. it can only sink current, not source current).

The LM324 is a quad operational amplifier – it’s designed for linear amplification and has push/pull outputs (i.e. it can source as well as sink current).

The two chips aren’t equivalent at all, and will behave very differently.

Reply:
The purpose of the LM339 is to output a square wave. I explained how the circuit works [here](https://medium.com/@enzo-lombardi/sounding-dribbling-episode-two-769fed9a7d8f). So for the purpose of squaring the wave, the LM324 is a great fit. I simulated both circuits and the difference are negligible.
Considering that a more accurate LM339 emulation is not an option for performance reason, I would say that taking the freedom to replace it with LM324 is ok since the OpAmp does go into saturation. Emulation in analog devices seems pretty common in netlist base code.

Reply:
No, an LM324 is *not* a great fit.  It will not work the same way at all.

Looking at the “STOP PALLA” circuit:
* With an LM339, the timing capacitor will charge via the 2.2kΩ and 10kΩ resistors in series an the LM339 has no high driver output.
* With an LM324, the timing capacitor will charge via the 10kΩ resistor only, the 2.2kΩ resistor will be effectively short-circuited by the high driver output of the LM324.

This will affect the frequency and symmetry of waveform.

An LM324 is also not a suitable replacement in the “PARATA” circuit, for more complex reasons.

If the LM339 model you used is too slow, you need a more efficient

Reply:
If the LM339 model you used has performance issues, the solution is to make a simplified LM339 model.  Note that the op-amps don’t use transistor-level models, they use high-level models found in src/lib/netlist/analog/nld_opamps.cpp

The key thing is that the LM339 **is not an op-amp**.  It’s a voltage comparator with a single-ended output.  Replacing it with an op-amp is not going to work.